src/topkg_jbuilder.ml

```diff
@@ -1,11 +1,57 @@
 open StdLabels
 open Topkg
+open Fpath
+
+let uerror = Unix.error_message
+
+let rec mkdir dir =
```

Comment:
@dbuenzli: any objection into moving `mkdir` and `copy` upstream into `Topkg.OS.Dir`?